### PR TITLE
[WIP] Fix creation of disks for RAID Test

### PR DESCRIPTION
**Pull request by DrMullings**
~~Partitioning for RAID test is now working, now fails in product bug:~~
~~https://bugzilla.suse.com/show_bug.cgi?id=1061865~~

Verification run SLE15:
http://10.160.65.204/tests/795
Verification run SLE12SP3:
http://10.160.65.204/tests/813#

Needles:
https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/521

### CODE REVIEW

File: tests/installation/partitioning_raid.pm [hunk 1]
```diff
@@ -211,51 +210,58 @@ sub run {
     }
 
     for (qw(vda vdb vdc vdd)) {
-        assert_screen "partitioning_raid-disk_$_-selected";
+        my $timeout = 2;
+        my $counter = 50;
```
Comment (rwx788) [inline tests/installation/partitioning_raid.pm]: Do we really need 50 steps to navigate to the disk? As we go in one direction only, probably we don't need that much, as if needle doesn't match, we will do a lot of unnecessary steps with 2 seconds timeout.

Comment (DrMullings) [reply tests/installation/partitioning_raid.pm]: I've tried different Settings here and those were the most reliable.
Might be that we can decrease it in the future (with a better qemu version) but for now I would keep it

**Comment by DrMullings** [comment]
According to ancor the bug is fixed in the next build, so maybe we should wait with merging and fix it alltogether

File: tests/installation/partitioning_raid.pm [hunk 2]
```diff
@@ -211,51 +210,58 @@ sub run {
...
             assert_screen 'partitioning_raid-mount_point_boot_efi';
             send_key $cmd{finish};
             assert_screen 'expert-partitioner';
             send_key 'shift-tab';
             send_key 'shift-tab';
-            send_key 'left';                                                                         # go to top "Hard Disks" node
+            send_key 'left';                                                                                            # go to top "Hard Disks" node
             assert_screen 'partitioning_raid-hard_disks-unfolded';
-            send_key 'left';                                                                         # fold the drive tree again
+            send_key 'left';                                                                                            # fold the drive tree again
```
Comment (okurz) [inline tests/installation/partitioning_raid.pm]: That comments on the right hand side are quite far apart. Can you try to put the comments on top of each line? Or try to add in another blank line before this section, e.g. line 230 and above, and try what `tools/tidy` gives you

**Comment by DrMullings** [comment]
Product bug got fixed, now test works on SLE12SP3 and theoretically on SLE15, here is just a verification run needed after I created the last needle.

File: tests/installation/partitioning_raid.pm [hunk 3]
```diff
@@ -304,38 +328,41 @@ sub run {
...
     # select file-system
     assert_screen 'partition-format';
-    send_key $cmd{filesystem};
+    send_key(is_storage_ng() ? 'alt-a' : $cmd{filesystem});
     assert_screen 'partitioning_raid-filesystem-focused';
-    send_key "end";
-    assert_screen 'partitioning_raid-swap_format-selected';
-    send_key $cmd{finish};
+    #send_key "end";
+    #assert_screen 'partitioning_raid-swap_format-selected';
```
Comment (okurz) [inline tests/installation/partitioning_raid.pm]: Don't add disabled code as nobody but you will understand why this is not active. Better leave it out and keep in your local git repository, either in `git stash` or a temporary "WIP"-commit

File: tests/installation/partitioning_raid.pm [hunk 4]
```diff
@@ -352,7 +379,8 @@ sub run {
     else {
         $needle_suffix = $needle_raid_swap_added_suffixes{uefi};
     }
-    assert_screen 'partitioning_raid-raid_swap_added' . $needle_suffix;
+    #assert_screen 'partitioning_raid-raid_swap_added' . $needle_suffix;
```
Comment (okurz) [inline tests/installation/partitioning_raid.pm]: dead code

**Comment by DrMullings** [comment]
Removed the commented code, still need verification runs that get to the end.
Please be aware that lost keys are also a problem in production

**Comment by DrMullings** [comment]
Updated SLE12SP3 run

File: tests/installation/partitioning_raid.pm [hunk 5]
```diff
@@ -211,52 +210,72 @@ sub run {
...
+        send_key "down" unless is_storage_ng;
     }
 
     # select RAID add
-    assert_screen 'partitioning_raid-raid-selected';
-    send_key $cmd{addraid};
+    if (is_storage_ng) {
+        send_key_until_needlematch 'partitioning_raid-raid-selected', 'down';
+        # hotkey is different on SLE15
+        send_key 'alt-d';
```
Comment (rwx788) [inline tests/installation/partitioning_raid.pm]: Let's redefine it like we did for donotformat, see https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/installation/partitioning.pm#L27

File: tests/installation/partitioning_raid.pm [hunk 6]
```diff
@@ -304,38 +328,39 @@ sub run {
...
+    send_key(is_storage_ng() ? 'alt-d' : $cmd{addraid});
     assert_screen 'partitioning_raid-menu_add_raid';
     setraidlevel(0);    # RAID0 for swap
     assert_screen 'partitioning_raid-raid_0-selected';
     addraid(1);
 
     # select file-system
     assert_screen 'partition-format';
-    send_key $cmd{filesystem};
+    send_key(is_storage_ng() ? 'alt-a' : $cmd{filesystem});
```
Comment (rwx788) [inline tests/installation/partitioning_raid.pm]: See comment above, if it makes sense to redefine, in case it's same button.

**Comment by rwx788** [comment]
Commit is part of https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/3723

**Pull request closed without merging**